codegenerator/cli/templates/static/codegen/src/Config.res

```diff
@@ -140,9 +140,9 @@ let shouldSaveHistory = (config, ~isInReorgThreshold) =>
   | _ => false
   }
 
-let shouldPruneHistory = config =>
+let shouldPruneHistory = (config, ~isInReorgThreshold) =>
   switch config.historyConfig {
-  | {historyFlag: MinHistory} => true
+  | {rollbackFlag: RollbackOnReorg, historyFlag: MinHistory} if isInReorgThreshold => true
```

Comment:
@DZakh note I made one more change here. I noticed on the monitoring that even the non-rollback indexers were firing this off. (Obviously a very light query with no history but still should only apply to rollback indexers)

And I baked in the isInReorgThreshold check here just like the shouldSaveHistory function above